sql: include ON UPDATE on CREATE TABLE LIKE for INCLUDING DEFAULTS

otan opened the pull request:
Resolves #69258

Release note (sql change): CREATE TABLE ... LIKE ... now copies ON
UPDATE definitions for INCLUDING DEFAULTS.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/70472)
<!-- Reviewable:end -->

rafiss commented on pkg/sql/logictest/testdata/logic_test/create_table:
```diff
@@ -169,9 +171,10 @@ like_constraints  CREATE TABLE public.like_constraints (
                   h INT8 NULL,
                   j JSONB NULL,
                   k INT8 NULL,
+                  t TIMESTAMPTZ NULL ON UPDATE current_timestamp():::TIMESTAMPTZ,
```
hm this table was made with `INCLUDING CONSTRAINTS` so it shouldn't have `ON UPDATE` right? (similar to how it doesn't have the `DEFAULT` on `b`

rafiss commented on pkg/sql/logictest/testdata/logic_test/create_table:
```diff
@@ -191,11 +194,12 @@ like_indexes  CREATE TABLE public.like_indexes (
               h INT8 NULL,
               j JSONB NULL,
               k INT8 NULL,
+              t TIMESTAMPTZ NULL ON UPDATE current_timestamp():::TIMESTAMPTZ,
```
also should not have ON UPDATE

rafiss commented on pkg/sql/logictest/testdata/logic_test/create_table:
```diff
@@ -212,9 +216,10 @@ like_generated  CREATE TABLE public.like_generated (
                 h INT8 NULL,
                 j JSONB NULL,
                 k INT8 NULL,
+                t TIMESTAMPTZ NULL ON UPDATE current_timestamp():::TIMESTAMPTZ,
```
also should not have ON UPDATE

rafiss commented on pkg/sql/logictest/testdata/logic_test/create_table:
```diff
@@ -230,9 +235,10 @@ like_defaults  CREATE TABLE public.like_defaults (
                h INT8 NULL,
                j JSONB NULL,
                k INT8 NULL,
+               t TIMESTAMPTZ NULL ON UPDATE current_timestamp():::TIMESTAMPTZ,
```
i'd be fine keeping this one. i vote we just decide now that `INCLUDING DEFAULTS` will also copy `ON UPDATE`. but then one thing to check is that a column with both `DEFAULT` and `ON UPDATE` is copied correctly

otan replied:
i've changed this to be for DEFAULTS because ALL is a bitmask meaning including it on every option. i'm a bad proofreader.

otan committed:
sql: include ON UPDATE on CREATE TABLE LIKE for INCLUDING DEFAULTS

Release note (sql change): CREATE TABLE ... LIKE ... now copies ON
UPDATE definitions for INCLUDING DEFAULTS.

rafiss reviewed (approved):
lgtm!

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70472#-:-Mk7nXwj9E-ZUQPl65jM:b-7nbn20)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=rafiss
